homeassistant/components/plugwise_stick/switch.py

```diff
@@ -0,0 +1,114 @@
+"""Support for Plugwise Circle(+) nodes."""
+from homeassistant.components.switch import SwitchEntity
+
+from . import PlugwiseNodeEntity
+from .const import (
+    AVAILABLE_SENSOR_ID,
+    CURRENT_POWER_SENSOR_ID,
+    DOMAIN,
+    SENSORS,
+    SWITCHES,
+    TODAY_ENERGY_SENSOR_ID,
+)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Plugwise switch based on config_entry."""
+    stick = hass.data[DOMAIN][entry.entry_id]["stick"]
+    nodes_data = hass.data[DOMAIN][entry.entry_id]["switch"]
+    entities = []
+    for mac in nodes_data:
+        node = stick.node(mac)
+        for switch_type in node.get_switches():
+            if switch_type in SWITCHES:
+                if (CURRENT_POWER_SENSOR_ID in node.get_sensors()) and (
+                    TODAY_ENERGY_SENSOR_ID in node.get_sensors()
+                ):
+                    entities.append(
+                        PlugwiseSwitchWithPower(
+                            node,
+                            mac,
+                            switch_type,
+                            CURRENT_POWER_SENSOR_ID,
+                            TODAY_ENERGY_SENSOR_ID,
+                        )
+                    )
+                else:
+                    entities.append(PlugwiseSwitch(node, mac, switch_type))
+    async_add_entities(entities)
+
+
+class PlugwiseSwitch(PlugwiseNodeEntity, SwitchEntity):
+    """Representation of a switch."""
+
+    def __init__(self, node, mac, switch_id):
+        """Initialize a Node entity."""
+        super().__init__(node, mac)
+        self.switch_id = switch_id
+        self.switch_type = SWITCHES[self.switch_id]
+        self.node_callbacks = (AVAILABLE_SENSOR_ID, self.switch_id)
+
+    @property
+    def entity_registry_enabled_default(self):
+        """Return the switch registration state."""
+        return self.switch_type["enabled_default"]
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self.switch_type["icon"]
```

Comment:
Something to check, we learned last week that when using DEVICE_CLASS you can't override icons, not sure if/how that applies to switches. https://developers.home-assistant.io/docs/core/entity/switch/#available-device-classes if you set it as device class then HA core will use default (and a user can always override themselves)

Reply:
Thanks for you comment. I fixed this in commit [a5fd45c](https://github.com/home-assistant/core/pull/35713/commits/a5fd45c74db2d3ef041b3095c34806899143293f)
While fixing this I've noticed some additional code simplification was possible too.